src/encoded/types/analyte.py

```diff
@@ -15,3 +35,64 @@ class Analyte(SubmittedItem):
     item_type = "analyte"
     schema = load_schema("encoded:schemas/analyte.json")
     embedded_list = []
+
+    class Collection(Item.Collection):
+        pass
+
+
+def validate_rna_molecule_properties(context,request):
+    """Check that `molecule` contains RNA if RNA-specific properties have values.
+    
+    RNA-specific properties in `RNA_PROPERTIES`.
+    """
+
+    data = request.json
+    if 'molecule' in data:
+        if 'RNA' not in data['molecule']:
+            for property in RNA_PROPERTIES:
+                if data.get(property,""):
+                    msg = f"Property {property} is specific to molecule RNA."
+                    return request.errors.add('body', 'Analyte: invalid property values', msg)
+                else:
```

Comment:
Very minor comment: Could leave this else off and fall through to the return request.validated.update({}) below.